### PR TITLE
feat(AnalyticalTable): Add prop selectionMode

### PR DESCRIPTION
**Pull request by MarcusNotheis**
Allows different selection modes via `TableSelectionMode` enum with values `NONE`, `SINGLE_SELECT`, `MULTI_SELECT`
BREAKING CHANGE: Replace prop `selectable` with `selectionMode`

### CODE REVIEW

File: packages/main/src/components/AnalyticalTable/hooks/useTableRowStyling.ts [hunk 1]
```diff
@@ -1,48 +1,47 @@
...
-      }
-      return rowProps;
-    });
-
-    return instance;
-  };
 
-  hook.pluginName = 'useTableRowStyling';
-  return hook;
+        console.log('Inner', selectionMode);
```
Comment (vbersch) [inline packages/main/src/components/AnalyticalTable/hooks/useTableRowStyling.ts]: please remove

File: packages/main/src/components/AnalyticalTable/types/ColumnType.ts [hunk 2]
```diff
@@ -1,10 +1,11 @@
-import { ComponentType } from 'react';
+import { ComponentType, RefObject } from 'react';
 import { Column } from 'react-table';
+import { Ui5PopoverDomRef } from '../../../interfaces/Ui5PopoverDomRef';
```
Comment (vbersch) [inline packages/main/src/components/AnalyticalTable/types/ColumnType.ts]: should we import from `@ui5/webcomponents-react/interfaces/UI5PopoverDomRef` or do we not care in this case?

File: packages/main/src/components/AnalyticalTable/virtualization/VirtualTableBody.tsx [hunk 3]
```diff
@@ -2,6 +2,7 @@ import '@ui5/webcomponents-icons/dist/icons/navigation-down-arrow';
 import '@ui5/webcomponents-icons/dist/icons/navigation-right-arrow';
 import React, { useCallback, useEffect, useMemo, useRef } from 'react';
 import { FixedSizeList } from 'react-window';
+import { TableSelectionMode } from '../../../lib/TableSelectionMode';
```
Comment (vbersch) [inline packages/main/src/components/AnalyticalTable/virtualization/VirtualTableBody.tsx]: should we import from `@ui5/webcomponents-react/lib/TableSelectionMode`?

Comment (MarcusNotheis) [reply packages/main/src/components/AnalyticalTable/virtualization/VirtualTableBody.tsx]: fixed

File: packages/main/src/components/AnalyticalTable/types/ColumnType.ts (hunk 2 above)
Comment (MarcusNotheis) [reply packages/main/src/components/AnalyticalTable/types/ColumnType.ts]: fixed

**Pull request merged**